connection-loader/src/main/java/com/terracotta/connection/api/DetailedConnectionException.java

```diff
@@ -0,0 +1,33 @@
+package com.terracotta.connection.api;
+
+import org.terracotta.connection.ConnectionException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/*
+  Should this class be going to connection-api project?
+   */
+public class DetailedConnectionException extends ConnectionException{
+
+  /**
+   * This type can only be created as a high-level wrapper over the underlying cause.
+   *
+   * @param cause The underlying throwable
+   * @param errorMap is the map of a host:port combination and list of all the communication errors for this host:port combination
+   */
+
+  private final Map<String, List<Exception>> connectionErrorMap;
```

Comment:
Exceptions in this map are implementation specific exceptions, , don't we need to create corresponding exceptions in connections api so that connection users can interpret these exceptions?